examples/Examples/Api.hs

```diff
@@ -0,0 +1,157 @@
+{-# LANGUAGE QualifiedDo #-}
+
+module Examples.Api (tests) where
+
+import Data.Proxy (Proxy (..))
+import Plutarch
+import Plutarch.Api.V1 (
+  PScriptContext (..),
+  PTxInInfo (..),
+  PTxInfo (..),
+  PValue (..),
+ )
+import Plutarch.Builtin (PAsData, PBuiltinList)
+import Plutarch.DataRepr (pindexDataList)
+import Plutarch.Lift (pconstant)
+import qualified Plutarch.Monadic as P
+import Plutarch.Trace (ptrace)
+
+import Plutus.V1.Ledger.Api (
+  Address (..),
+  Credential (..),
+  CurrencySymbol,
+  DatumHash,
+  ScriptContext (..),
+  ScriptPurpose (..),
+  TxInInfo (..),
+  TxInfo (..),
+  TxOut (..),
+  TxOutRef (..),
+  ValidatorHash,
+  Value,
+ )
+import qualified Plutus.V1.Ledger.Interval as Interval
+import qualified Plutus.V1.Ledger.Value as Value
+import Test.Tasty (TestTree, testGroup)
+import Test.Tasty.HUnit (testCase)
+
+import Utils
+
+--------------------------------------------------------------------------------
+
+{- |
+  An example 'PScriptContext' Term,
+  lifted with 'pconstant'
+-}
+ctx :: Term s PScriptContext
+ctx =
+  pconstant
+    (ScriptContext info purpose)
+
+-- | Simple script context, with minting and a single input
+info :: TxInfo
+info =
+  TxInfo
+    { txInfoInputs = [inp]
+    , txInfoOutputs = []
+    , txInfoFee = mempty
+    , txInfoMint = mint
+    , txInfoDCert = []
+    , txInfoWdrl = []
+    , txInfoValidRange = Interval.always
+    , txInfoSignatories = []
+    , txInfoData = []
+    , txInfoId = "b0"
+    }
+
+-- | A script input
+inp :: TxInInfo
+inp =
+  TxInInfo
+    { txInInfoOutRef = ref
+    , txInInfoResolved =
+        TxOut
+          { txOutAddress =
+              Address (ScriptCredential validator) Nothing
+          , txOutValue = mempty
+          , txOutDatumHash = Just datum
+          }
+    }
+
+-- | Minting a single token
+mint :: Value
+mint = Value.singleton sym "sometoken" 1
+
+ref :: TxOutRef
+ref = TxOutRef "a0" 0
+
+purpose :: ScriptPurpose
+purpose = Spending ref
+
+validator :: ValidatorHash
+validator = "a1"
+
+datum :: DatumHash
+datum = "d0"
+
+sym :: CurrencySymbol
+sym = "c0"
+
+--------------------------------------------------------------------------------
+
+_getTxInfo :: Term s (PScriptContext :--> PAsData PTxInfo)
+_getTxInfo = plam $ \x -> P.do
+  PScriptContext c <- pmatch x
+  pindexDataList (Proxy @0) # c
+
+_getMint :: Term s (PTxInfo :--> PAsData PValue)
+_getMint = plam $ \x -> P.do
+  PTxInfo i <- pmatch x
+  pindexDataList (Proxy @3) # i
+
+_getInputs :: Term s (PTxInfo :--> PAsData (PBuiltinList (PAsData PTxInInfo)))
+_getInputs = plam $ \x -> P.do
+  PTxInfo i <- pmatch x
+  ptrace "xhuawdhauywhd"
+  i' <- plet i
+  pindexDataList (Proxy @0) # i'
+
+{-
+-- | Get first validator from TxInInfo
+getValidator :: Term s (PBuiltinList (PAsData PTxInInfo) :--> PAsData PValidatorHash)
```

Comment:
I think we can add this back now.

Reply:
Partially resolved by #158

Reply:
Further progress by #161 
Possibly fully resolved by #159